### PR TITLE
Remove the tracing lambda from IO.apply

### PR DESCRIPTION
**Pull request by vasilmkd**
Curious to see if the deadlock that arose during the original development can still be seen.

### CODE REVIEW

**Comment by vasilmkd** [comment]
https://discord.com/channels/632277896739946517/839263556754472990/875499045085384755

File: core/shared/src/main/scala/cats/effect/IO.scala [hunk 1]
```diff
@@ -929,10 +929,8 @@ object IO extends IOCompanionPlatform with IOLowPriorityImplicits {
    *
    * Alias for `IO.delay(body)`.
    */
-  def apply[A](thunk: => A): IO[A] = {
-    val fn = () => thunk
-    Delay(fn, Tracing.calculateTracingEvent(fn.getClass))
-  }
+  def apply[A](thunk: => A): IO[A] =
+    Delay(() => thunk, Tracing.calculateTracingEvent(thunk.getClass))
```
Comment (armanbilge) [inline core/shared/src/main/scala/cats/effect/IO.scala:933]: I've been surprised before, but this isn't really evaluating the `thunk` first, then calling `getClass` ?

Comment (armanbilge) [reply core/shared/src/main/scala/cats/effect/IO.scala:933]: And if not, then _how_ do you call `getClass` on the outcome of the `thunk`?

**Comment by vasilmkd** [comment]
The CI is hopelessly stuck.

File: core/shared/src/main/scala/cats/effect/IO.scala (hunk 1 above)
Comment (armanbilge) [reply core/shared/src/main/scala/cats/effect/IO.scala:933]: e.g. see https://scastie.scala-lang.org/Gq9MqhQoS9KPSGZQjV2kNA

**Comment by vasilmkd** [comment]
Fixed in #2226.

**Pull request closed without merging**